namespace protocols { namespace common {

enum ErrorCode
{
	EC_SUCCESS = 0;
	EC_INVALID_REQUEST = 1;
	EC_INVALID_ARGUMENT = 2;
	EC_IO_ERROR = 3;
	EC_NET_ERROR = 4;
	EC_DB_ERROR = 5;
	EC_ACTION_DENY = 6;

} ;

enum MessageType {
  MSG_TYPE_REQUEST = 0,
  MSG_TYPE_RESPONSE = 1,
  MSG_TYPE_NOTIFY = 2
};

enum MessageAction {
  MSG_ACTION_ACC_LOGIN = 1,
  MSG_ACTION_PLAYER_LOGIN = 2,
  MSG_ACTION_INIT_CONNECTION = 3,
  
  MSG_ACTION_GET_ACCOUNT_LIST = 4,
  MSG_ACTION_MOVE_START = 5,
  MSG_ACTION_CREATE_PLAYER = 7,
  MSG_ACTION_PLAYER_INFO = 8,
  
  MSG_ACTION_BUDDY_DETAIL_INFO = 9,
  MSG_ACTION_HEARTBEAT = 10,
  MSG_ACTION_KICK_PLAYER = 11,
  
};

enum ENUM_ATTR {
  NONE = 0,
  HP = 1,
  MAX_HP = 2,
  STA = 3,
  STRENGTH = 4,
  TECHNIQUE = 5,
  INTELLIGENCE = 6,
  SPIRIT = 7,
  AGILE = 8,
  PHY_ATTACK = 9,
  PHY_DEFENCE = 10,
  MAG_ATTACK = 11,
  MAG_DEFENCE = 12,
  PHY_AT_HIT = 13,
  MAG_AT_HIT = 14,
  PHY_CRITICAL = 15,
  PHY_CRITICAL_DEFENCE = 16,
  MAG_CRITICAL = 17,
  MAG_CRITICAL_DEFENCE = 18,
  PHY_AT_DODGE = 19,
  MAG_AT_DODGE = 20,
  PHY_AT_HIT_RATE = 21,
  MAG_AT_HIT_RATE = 22,
  PHY_CRITICAL_RATE = 23,
  PHY_CRITICAL_DEFENCE_RATE = 24,
  MAG_CRITICAL_RATE = 25,
  MAG_CRITICAL_DEFENCE_RATE = 26,
  PHY_AT_DODGE_RATE = 27,
  MAG_AT_DODGE_RATE = 28,
  SPEED = 29,
  FIRE_RESISTANCE = 30,
  ICE_RESISTANCE = 31,
  SHOCK_RESISTANCE = 32,
  FIRE_DAMAGE_INCREASE = 33,
  ICE_DAMAGE_INCREASE = 34,
  SHOCK_DAMAGE_INCREASE = 35,
  PHY_DAMAGE_REDUCTION_RATE = 36,
  PHY_CRI_MULTI_RATE = 37,
  MAG_CRI_MULTI_RATE = 38,
  MAG_COMBO_RATE = 39,
  COUNTER_ATTACK_RATE = 40,
  DAMAGE_REBOUND_RATE = 41,
  BLOOD_SUCK = 42,
  
  ENUM_ATTR_NUMBER = 64
};



}
};